xtask/src/pre_cache.rs

```diff
@@ -1,79 +0,0 @@
-use std::{
-    fs::FileType,
-    path::{Path, PathBuf},
-};
-
-use anyhow::Result;
-use xshell::rm_rf;
-
-use crate::flags;
-
-impl flags::PreCache {
-    /// Cleans the `./target` dir after the build such that only
-    /// dependencies are cached on CI.
-    pub(crate) fn run(self) -> Result<()> {
-        let slow_tests_cookie = Path::new("./target/.slow_tests_cookie");
-        if !slow_tests_cookie.exists() {
-            panic!("slow tests were skipped on CI!")
```

Comment:
I wonder if it's worth keeping this check.

Reply:
https://github.com/rust-analyzer/rust-analyzer/blob/fc9eed4836dfc88fe2893c81b015ab440cea2ba6/crates/test_utils/src/lib.rs#L285
so essentially check that CI really defines the `CI` env var?

Reply:
Yeah, it's a check to make sure we set those two variables. We could test for that it in workflow, but it badly needs a refactoring (I know it's possible to remove the duplication, but I don't know the syntax).

Reply:
:thinking: I guess we can just remove this. If, at some point we notice that slow tests are not being run, we'll learn that the check was helpful in the end :)